services/dedup/mirrorBadger/mirrorBadger_test.go

```diff
@@ -0,0 +1,69 @@
+package mirrorBadger
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource/scylla"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/rand"
+	"github.com/rudderlabs/rudder-server/services/dedup/types"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+func Test_MirrorBadger(t *testing.T) {
+	config.Reset()
+	logger.Reset()
+	misc.Init()
+
+	dbPath := os.TempDir() + "/dedup_test"
```

Comment:
```suggestion
	dbPath := t.TempDir() + "/dedup_test"
```

Reply:
It checks for existence as well so you could remove the `_ = os.RemoveAll(dbPath)` that you have on line 27.
It should also do the cleanup for you if I'm not mistaken:

```go
		c.tempDir, c.tempDirErr = os.MkdirTemp("", pattern)
		if c.tempDirErr == nil {
			c.Cleanup(func() {
				if err := removeAll(c.tempDir); err != nil {
					c.Errorf("TempDir RemoveAll cleanup: %v", err)
				}
			})
		}
```